bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/gems/GemsUtils4Java.scala

```diff
@@ -1,17 +1,91 @@
 import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.shared.skyobject
+import edu.gemini.skycalc
+import scala.math._
+import scala.collection.JavaConverters._
+
+import scalaz._
+import Scalaz._
 
 /**
  * Utility methods for Java classes to access scala classes/methods
  */
 object GemsUtils4Java {
   // Returns true if the target magnitude is within the given limits
-  def containsMagnitudeInLimits(target: SPTarget, magLimits: MagnitudeConstraints): Boolean =
-    // TODO This method has too many conversions, should be simplified
-    target.getMagnitude(magLimits.band.toOldModel).asScalaOpt.map(m => magLimits.contains(m.toNewModel)).getOrElse(true)
+  def containsMagnitudeInLimits(target: SiderealTarget, magLimits: MagnitudeConstraints): Boolean =
+    target.magnitudeIn(magLimits.band).map(m => magLimits.contains(m)).getOrElse(true)
+
+  def mapMagnitudes(constraint: Option[SaturationConstraint], mapOp: immutable.MapOp[SaturationConstraint, java.lang.Double]): immutable.Option[java.lang.Double] =
+    constraint.map(m => mapOp.apply(m)).asGeminiOpt
+
+  // Combines multiple radius limits into one
+  def optimizeRadiusConstraint(criterList: java.util.List[GemsCatalogSearchCriterion]): RadiusConstraint = {
+    val result = criterList.asScala.foldLeft((Double.MinValue, Double.MaxValue)) { (prev, current) =>
+      val c = current.criterion
+      val radiusConstraint = c.adjustedLimits
+      val maxLimit = radiusConstraint.maxLimit
+      val correctedMax = (c.offset |@| c.posAngle) { (o, _) =>
+          // If an offset and pos angle were defined, normally an adjusted base position
+          // would be used, however since we are merging queries here, use the original
+          // base position and adjust the radius limits
+          maxLimit + o.distance
+        } | maxLimit
+      (max(correctedMax.toDegrees, prev._1), min(radiusConstraint.minLimit.toDegrees, prev._2))
+    }
+    RadiusConstraint.between(Angle.fromDegrees(result._1), Angle.fromDegrees(result._2))
+  }
+
+  /**
+   * Sorts the targets list, putting the brightest stars first and returns the sorted array.
+   */
+  def sortTargetsByBrightness(targetsList: java.util.List[SiderealTarget]): java.util.List[SiderealTarget] =
+    targetsList.asScala.sortBy(_.magnitudeIn(MagnitudeBand.R)).asJava
+
+  // TODO Star should be replaced by SiderealTarget
+  def starToSiderealTarget(star: Star): SiderealTarget = {
+    val ra = RightAscension.fromAngle(Angle.fromDegrees(star.ra))
+    val dec = Declination.fromAngle(Angle.fromDegrees(star.dec)).getOrElse(Declination.zero)
+    val coords = Coordinates(ra, dec)
+
+    val magnitudes = List((star.bmag, MagnitudeBand.B), (star.vmag, MagnitudeBand.V), (star.rmag, MagnitudeBand.R), (star.jmag, MagnitudeBand.J), (star.hmag, MagnitudeBand.H), (star.kmag, MagnitudeBand.K))
+
+    val invalid = MascotConf.invalidMag
+    val mags = magnitudes.filter(_._1 != invalid).map(v => new Magnitude(v._1, v._2))
```

Comment:
When I see `x.filter(...).map` I think `collect`.  It doesn't get much shorter but debatably easier to read without the `._1`s.

````scala
magnitudes.collect { case (mag, band) if mag != invalid => new Magnitude(mag, band) }
````